backend/Squirrel.Core/Squirrel.Core.BLL/Services/UserService.cs

```diff
@@ -192,4 +203,44 @@ private User PrepareNewUserData(UserRegisterDto userDto, bool isGoogleAuth)
 
         return newUser;
     }
+
+    public async Task AddAvatar(IFormFile avatar)
+    {
+        var userEntity = await GetUserByIdInternal(_userIdGetter.GetCurrentUserId());
+
+        var content = await CropAvatar(avatar);
+
+        var blob = new Blob
+        {
+            Id = userEntity.Id.ToString(),
+            ContentType = avatar.ContentType,
+            Content = content
+        };
+
+        var url = await _blobStorageService.UploadWithUrlAsync("user-avatars", blob);
+        userEntity.AvatarUrl = url;
+        await _context.SaveChangesAsync();
+    }
+
+    public async Task DeleteAvatar()
+    {
+        var userEntity = await GetUserByIdInternal(_userIdGetter.GetCurrentUserId());
+        await _blobStorageService
+            .DeleteAsync("user-avatars", userEntity.Id.ToString());
+
+        userEntity.AvatarUrl = String.Empty;
+        await _context.SaveChangesAsync();
+    }
+
+    private async Task<byte[]> CropAvatar(IFormFile avatar)
+    {
+        using var image = await Image.LoadAsync(avatar.OpenReadStream());
```

Comment:
we should check the image size before load

for example, up to 5MB is allowed, but upper - prohibited and throw error

Reply:
good point